app/models/participant.rb

```diff
@@ -1,9 +1,13 @@
 class Participant < ActiveRecord::Base
   has_many :credits, dependent: :destroy
+  has_many :posts, through: :credits
 
   belongs_to :site
 
   validates :name, presence: true
 
+  scope :with_published_posts, -> { joins(:posts).merge(Post.published).uniq }
```

Comment:
I feel that the scope name `with_published_posts` means `Post.published` is eager loaded.